fedgov-cv-lib-asn1c/src/main/java/rga_wrapper.c

```diff
@@ -20,29 +20,261 @@
+
+	jint year = (*env)->CallIntMethod(env, timeOfCalcObj, getYear);
+	jint month = (*env)->CallIntMethod(env, timeOfCalcObj, getMonth);
+	jint day = (*env)->CallIntMethod(env, timeOfCalcObj, getDay);
+
+	timeOfCalculation.year = (long)year;
+	timeOfCalculation.month = (long)month;
+	timeOfCalculation.day = (long)day;
+
+	refPointInfo.timeOfCalculation = timeOfCalculation;
+
+	rgaBaseLayer.refPointInfo = refPointInfo;
+
+	// Road Geometry Ref ID Info (relativeToRdAuthID)
+	RoadGeometryRefIDInfo_t rdGeoRefID;
+
+	RoadAuthorityID_t *roadAuthorityID = calloc(1, sizeof(RoadAuthorityID_t));
+
+	// Check if full RAID exists
+	jmethodID isFullRdAuthIDExists = (*env)->GetMethodID(env, baseLayerClass, "isFullRdAuthIDExists", "()Z");
+	jboolean fullRdAuthIDExists = (*env)->CallBooleanMethod(env, baseLayer, isFullRdAuthIDExists);
+
+
+	// Check if relative RAID exists
+	jmethodID isRelRdAuthIDExists = (*env)->GetMethodID(env, baseLayerClass, "isRelRdAuthIDExists", "()Z");
+	jboolean relRdAuthIDExists = (*env)->CallBooleanMethod(env, baseLayer, isRelRdAuthIDExists);
+	
+	if (fullRdAuthIDExists)
+	{
+		// Get full RAID
+		jmethodID getFullRdAuthID = (*env)->GetMethodID(env, baseLayerClass, "getFullRdAuthID", "()[I");
+		jintArray fullRdAuthID = (*env)->CallObjectMethod(env, baseLayer, getFullRdAuthID);
+		const int *fullRdAuthIDStr = (*env)->GetIntArrayElements(env, fullRdAuthID, NULL);
+
+		jsize len = (*env)->GetArrayLength(env, fullRdAuthID);
+		size_t fullRdAuthIDStrLen = len;
+
+		ASN__PRIMITIVE_TYPE_t primType;
+		primType.buf = (uint8_t *)calloc(1, (fullRdAuthIDStrLen + 1));
+
+		for (size_t n = 0; n < fullRdAuthIDStrLen; n++)
+		{
+			primType.buf[n] = (uint8_t)fullRdAuthIDStr[n];
+		}
+		primType.size = fullRdAuthIDStrLen;
+
+		// Set RAID pointer's PRESENT and CHOICE
+		roadAuthorityID->present = RoadAuthorityID_PR_fullRdAuthID;
+		roadAuthorityID->choice.fullRdAuthID = primType;
+		rdGeoRefID.rdAuthorityID = roadAuthorityID;
+	} else if (relRdAuthIDExists) {
+		// Get relative RAID
+		jmethodID getRelRdAuthID = (*env)->GetMethodID(env, baseLayerClass, "getRelRdAuthID", "()[I");
+		jintArray relRdAuthID = (*env)->CallObjectMethod(env, baseLayer, getRelRdAuthID);
+
+		const int *relRdAuthIDStr = (*env)->GetIntArrayElements(env, relRdAuthID, NULL);
+
+		jsize len  = (*env)->GetArrayLength(env, relRdAuthID);
+
+		size_t relRdAuthIDStrLen = len;
+		ASN__PRIMITIVE_TYPE_t primType;
+		primType.buf = (uint8_t *)calloc(1, (relRdAuthIDStrLen + 1));
+
+		for (size_t n = 0; n < relRdAuthIDStrLen; n++)
+		{
+			primType.buf[n] = (uint8_t)relRdAuthIDStr[n];
+		}
+		primType.size = relRdAuthIDStrLen;
+
+		// Set RAID pointer's PRESENT and CHOICE
+		roadAuthorityID->present = RoadAuthorityID_PR_relRdAuthID;
+		roadAuthorityID->choice.relRdAuthID = primType;
+		rdGeoRefID.rdAuthorityID = roadAuthorityID;
+	} else {
+		roadAuthorityID->present = RoadAuthorityID_PR_NOTHING;
+	}
+	//rdGeoRefID.rdAuthorityID = roadAuthorityID;
+	rdGeoRefID.rdAuthorityID = NULL;
```

Comment:
It should, however, since I've changed RAID in the RGA (Encoder and JNI Wrapper) to int[] the message builder is throwing errors (since the message builder and MAP encoder are still implemented with String types). 

I'm setting it to NULL overall right now until the implementation is changed in the message builder and with the MAP Encoder.

I can write a TODO to detail this.

Reply:
yes, please add a TODO there to mention it is temporary. But also set it to null within the else condition if it applies .